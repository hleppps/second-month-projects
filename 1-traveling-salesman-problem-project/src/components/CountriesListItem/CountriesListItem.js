import { useContext } from "react";
import CountriesContext from "../../store/countries-context";
import styles from "./CountriesListItem.module.css";

const CountriesListItem = (props) => {
  const { dragItem } = useContext(CountriesContext);

  const selectCountryHandler = () => {
    props.onSelectCountry(props.countryName);
  };

  const dragStartHandler = (event, item, list) => {
    dragItem.set({ country: item, type: list });
  };

  const dragEndHandler = () => {
    dragItem.set({});
  };

  return (
    <li
      data-testid="countryListItem"
      className={styles.listItem}
      onClick={selectCountryHandler}
      draggable={true}
      onDragStart={(event) =>
        dragStartHandler(event, props.countryName, props.listType)
      }
      onDragEnd={() => dragEndHandler()}
    >
      <p data-testid="countryName" className={styles.name}>
        {props.countryName}
      </p>
      <img
        data-testid="flag"
        className={styles.flag}
        src={props.flagUrl}
        alt={props.flagUrl}
        draggable={false}
      />
    </li>
  );
};

export default CountriesListItem;
