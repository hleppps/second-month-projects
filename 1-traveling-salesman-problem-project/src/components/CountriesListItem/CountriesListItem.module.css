@import url(../../_variables.css);

.listItem {
  user-select: none;
  cursor: pointer;
  margin: 0.5rem 0;
  padding: 0.4rem;
  border: 1px solid var(--selected-text-border-color);
  border-radius: 1rem;
  color: var(--selected-text-color);
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.name {
  margin: 0;
  width: 80%;
  text-align: left;
  display: inline-block;
}

.flag {
  width: 18%;
  max-width: 30px;
  height: 20px;
  object-fit: cover;
  border: 1px solid #3b2403;
  border-radius: .5rem;
}