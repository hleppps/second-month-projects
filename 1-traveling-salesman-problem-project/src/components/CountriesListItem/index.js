import CountriesListItem from "./CountriesListItem"

export default CountriesListItem