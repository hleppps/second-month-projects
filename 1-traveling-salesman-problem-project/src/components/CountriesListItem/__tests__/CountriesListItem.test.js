import "@testing-library/jest-dom";
import "@testing-library/jest-dom/extend-expect";
import { cleanup, fireEvent, render } from "@testing-library/react";
import React from "react";
import CountriesContext from "../../../store/countries-context";
import CountriesListItem from "../CountriesListItem";

const renderWithContext = (component, providerProps) => {
  return {
    ...render(
      <CountriesContext.Provider value={providerProps}>
        {component}
      </CountriesContext.Provider>
    ),
  };
};

afterEach(cleanup);

describe("CountriesListItem", () => {
  it("renders with correct values from props", () => {
    const countryName = "Country";
    const flagUrl = "https://";

    const { getByTestId, getByAltText } = render(
      <CountriesListItem countryName={countryName} flagUrl={flagUrl} />
    );

    expect(getByTestId("countryName")).toHaveTextContent(countryName);
    getByAltText(flagUrl);
  });

  it("responds to click", () => {
    const onClickHandler = jest.fn();
    const { getByTestId } = render(
      <CountriesListItem onSelectCountry={onClickHandler} />
    );

    fireEvent.click(getByTestId("countryListItem"));

    expect(onClickHandler).toBeCalled();
  });

  it("responds to drag start", () => {
    const onDragHandler = jest.fn();
    const providerProps = {
      dragItem: { set: onDragHandler },
    };

    const { getByTestId } = renderWithContext(
      <CountriesListItem />,
      providerProps
    );

    fireEvent.dragStart(getByTestId("countryListItem"));
    expect(onDragHandler).toBeCalled();
  });

  it("matches a snapshot", () => {
    const { asFragment } = render(<CountriesListItem />);
    expect(asFragment(<CountriesListItem />)).toMatchSnapshot();
  });
});
