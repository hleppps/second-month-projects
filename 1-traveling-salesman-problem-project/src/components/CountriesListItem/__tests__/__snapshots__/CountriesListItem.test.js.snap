// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountriesListItem matches a snapshot 1`] = `
<DocumentFragment>
  <li
    class="listItem"
    data-testid="countryListItem"
    draggable="true"
  >
    <p
      class="name"
      data-testid="countryName"
    />
    <img
      class="flag"
      data-testid="flag"
      draggable="false"
    />
  </li>
</DocumentFragment>
`;
